Makefile

```
@@ -6,95 +6,211 @@ ifneq ($(OS),Windows_NT)
+	uv run pyright
+endif
 
 .PHONY: lint-shellcheck
-lint-shellcheck:
-	sh -c 'git ls-files | file --mime-type -Nnf- | grep shellscript | rev | cut -d: -f2- | rev | xargs -r shellcheck'
+lint-shellcheck: install-shellcheck  ##- Lint shell scripts
+	git ls-files | file --mime-type -Nnf- | grep shellscript | cut -f1 -d: | xargs -r shellcheck
 
 .PHONY: lint-yaml
-lint-yaml:  ## Lint YAML files with yamllint
-	yamllint .
+lint-yaml:  ##- Lint YAML files with yamllint
+	uv run --extra lint yamllint .
+
+.PHONY: lint-docs
+lint-docs:  ##- Lint the documentation
+	uv run --extra docs sphinx-lint --max-line-length 88 --enable all $(DOCS)
+
+---------------- : ## ----------------
+
+.PHONY: test
+test: test-unit test-integration  ## Run all tests
 
 .PHONY: test-unit
-test-unit: ## Run unit tests
-	uv run --frozen pytest --cov=$(PROJECT) --cov-config=pyproject.toml --cov-report=xml:.coverage.unit.xml --junit-xml=.results.unit.xml tests/unit
+test-unit:  ##- Run unit tests
+	uv run pytest tests/unit
 
 .PHONY: test-integration
-test-integration:  ## Run integration tests
-	uv run --frozen pytest --cov=$(PROJECT) --cov-config=pyproject.toml --cov-report=xml:.coverage.integration.xml --junit-xml=.results.integration.xml tests/integration
+test-integration:  ##- Run integration tests
+	uv run pytest tests/integration
+
+.PHONY: test-coverage
+test-coverage:  ## Generate coverage report
+	uv run coverage run --source $(PROJECT) -m pytest tests
+	uv run coverage xml -o coverage.xml
+	uv run coverage report -m
+	uv run coverage html
+
+---------------- : ## ----------------
+
+.PHONY: docs
+docs:  ## Build documentation
+	uv run --extra docs sphinx-build -b html -W docs docs/_build
+
+.PHONY: docs-auto
+docs-auto:  ## Build and host docs with sphinx-autobuild
+	uv run --extra docs sphinx-autobuild -b html --open-browser --port=8080 --watch $(PROJECT) -W docs docs/_build
+
+---------------- : ## ----------------
+
+.PHONY: package
+package: package-pip  ## Build all packages
+
+.PHONY: package-pip
+package-pip:  ##- Build packages for pip (sdist, wheel)
+	uv run pyproject-build --installer uv .
+
+---------------- : ## ----------------
+
+.PHONY: clean
+clean:  ## Clean up the development environment
+	uv tool run pyclean .
+	rm -rf dist/ build/ docs/_build/ *.snap .coverage*
+
+---------------- : ## ----------------
+
+# Below are intermediate targets for setup. They are not included in help as they should
+# not be used independently.
+
+.PHONY: install-uv
+install-uv:
+ifneq ($(shell which uv),)
+else ifneq ($(shell which snap),)
+	sudo snap install --classic astral-uv
+else ifneq ($(shell which brew),)
+	brew install uv
+else ifeq ($(OS),Windows_NT)
+	powershell -c "irm https://astral.sh/uv/install.ps1 | iex"
+else
+	curl -LsSf https://astral.sh/uv/install.sh | sh
+endif
+
+.PHONY: install-venv
+install-venv:
+ifneq ($(shell which apt-get),)
+	python3 -m venv --help && sudo apt-get --yes install python3-venv
```

Is it a good operator ? It seems to me as it should be `or`
```suggestion
	python3 -m venv --help || sudo apt-get --yes install python3-venv
```

Btw. is there still code that requires a normal "venv" ?

---

The starbase `test_setuptools.py` test does.